### PR TITLE
Add `XRPLNetwork` Enum

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

This PR pulls down [`XrplNetwork`](https://github.com/xpring-eng/Xpring-JS/blob/master/src/Common/xrpl-network.ts) into our common core library. 

### Context of Change

We often need to know what network components are connected to. This is best represented as an enum since there are multiple states. This enum is used liberally at the SDK library level (xpring-js, xpringkit, xpring4j) and we should use it here as well. 

Also add:
- A utility function to map a given `XrplNetwork` to whether it represents a test network
- Migrate `WalletFactory` to use this enum

The `WalletFactory` change is breaking, but no client libraries have consumed or exposed this class yet so I think it's not a big deal to change to it at this time. 

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

Networks are represented deterministically, code is more readable. 

## Test Plan

CI; additional tests provided for new methods.

## Future Tasks
- `XRPLNetwork` in Xpring-JS can be removed

### CODE REVIEW

File: src/XRP/wallet-factory.ts [hunk 1]
```diff
@@ -1,26 +1,32 @@
...
+  public constructor(network: XrplNetwork) {
+    this.network = network
+    this.isTest = XrpUtils.isTestNetwork(network)
   }
 
   /**
    * Generate a new wallet with the given keys.
    *
-   * @param publicKey - A hexadecimal encoded public key.
+   * @param publicKey - A hexadencimal encoded public key.
```
Comment (amiecorso) [inline src/XRP/wallet-factory.ts:29]:
```suggestion
   * @param publicKey - A hexadecimal encoded public key.
```

File: src/XRP/xrpl-network.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/**
+ * XRP Ledger networks components can be attached to.
```
Comment (amiecorso) [inline src/XRP/xrpl-network.ts:2]:
```suggestion
 * XRP Ledger networks that components can be attached to.
```

File: test/XRP/wallet-factory.test.ts [hunk 3]
```diff
@@ -2,11 +2,12 @@ import { assert } from 'chai'
 
 import WalletFactory from '../../src/XRP/wallet-factory'
 import 'mocha'
+import XRPLNetwork from '../../src/XRP/xrpl-network'
```
Comment (amiecorso) [inline test/XRP/wallet-factory.test.ts:5]: should the casing here be `XrplNetwork` ?

File: test/XRP/wallet-factory.test.ts [hunk 4]
```diff
@@ -2,11 +2,12 @@ import { assert } from 'chai'
...
 
 describe('Wallet Factory', function (): void {
   it('Wallet From Keys - Valid Keys', function (): void {
     // GIVEN a wallet factory and set of valid keys.
-    const walletFactory = new WalletFactory(true)
+    const walletFactory = new WalletFactory(XRPLNetwork.Test)
```
Comment (amiecorso) [inline test/XRP/wallet-factory.test.ts:10]: same for these three instances... `XrplNetwork`?

File: test/XRP/wallet-factory.test.ts (hunk 3 above)
Comment (keefertaylor) [reply test/XRP/wallet-factory.test.ts]: Old habits die hard.

**Pull request merged**